error: cannot construct `msghdr` with struct literal syntax due to private fields
   --> .../udevrs-0.3.0/src/monitor.rs:928:24
    |
928 |         let mut smsg = libc::msghdr {
    |                        ^^^^^^^^^^^^
    |
    = note: ...and other private fields `__pad1` and `__pad2` that were not provided

---

--- a/src/monitor.rs
+++ b/src/monitor.rs
@@ -925,15 +925,7 @@ impl UdevMonitor {
             },
         ];
 
-        let mut smsg = libc::msghdr {
-            msg_iov: iov.as_mut_ptr() as *mut _,
-            msg_iovlen: iov.len().try_into().unwrap(),
-            msg_control: core::ptr::null_mut(),
-            msg_controllen: 0,
-            msg_flags: 0,
-            msg_name: core::ptr::null_mut(),
-            msg_namelen: 0,
-        };
+        let mut smsg = unsafe { std::mem::zeroed::<libc::msghdr>() };
 
         if device.get_properties_monitor_buf().len() < 32 {
             Err(Error::UdevMonitor(
